ccl/crosscluster/logical: TestRandomStream failed

ccl/crosscluster/logical.TestRandomStream [failed](https://tanzanite.cluster.engflow.com/invocations/default/7e63dfa1-0f62-42fe-af9b-304563d6285d?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2Nyb3NzY2x1c3Rlci9sb2dpY2FsOmxvZ2ljYWxfdGVzdA==) on master @ [7917e91eadb2f622faadf635a3e13c384e42070e](https://github.com/cockroachdb/cockroach/commits/7917e91eadb2f622faadf635a3e13c384e42070e):


```
        created by github.com/sasha-s/go-deadlock.lock in goroutine 16840418
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182 +0x14d
        
        goroutine 13142405 [select]:
        github.com/cockroachdb/cockroach/pkg/sql/gcjob/gcjobnotifier.(*Notifier).run(0xc0191904e0, {0xc008206460?, 0xc88985?})
        	github.com/cockroachdb/cockroach/pkg/sql/gcjob/gcjobnotifier/notifier.go:145 +0xc8
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
        	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484 +0x13a
        created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 13084533
        	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:475 +0x3fe
        
        goroutine 16840112 [runnable]:
        github.com/sasha-s/go-deadlock.lock.func1()
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182
        created by github.com/sasha-s/go-deadlock.lock in goroutine 16840018
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182 +0x14d
        
        goroutine 16842010 [runnable]:
        github.com/sasha-s/go-deadlock.lock.func1()
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182
        created by github.com/sasha-s/go-deadlock.lock in goroutine 16841820
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182 +0x14d
        
        goroutine 16842281 [runnable]:
        github.com/sasha-s/go-deadlock.lock.func1()
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182
        created by github.com/sasha-s/go-deadlock.lock in goroutine 13403569
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182 +0x14d
        
        goroutine 16842259 [runnable]:
        github.com/sasha-s/go-deadlock.lock.func1()
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182
        created by github.com/sasha-s/go-deadlock.lock in goroutine 13403569
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182 +0x14d
        
        goroutine 16842815 [runnable]:
        github.com/sasha-s/go-deadlock.lock.func1()
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182
        created by github.com/sasha-s/go-deadlock.lock in goroutine 16842696
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182 +0x14d
        
        goroutine 16842809 [runnable]:
        github.com/sasha-s/go-deadlock.lock.func1()
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182
        created by github.com/sasha-s/go-deadlock.lock in goroutine 16842696
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182 +0x14d
    logical_replication_job_test.go:622: timed out waiting for events
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestRandomStream2201156389
--- FAIL: TestRandomStream (56.08s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomStream.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41024

ccl/crosscluster/logical.TestRandomStream [failed](https://tanzanite.cluster.engflow.com/invocations/default/cde783ce-0b25-4c2e-bbd9-428395867664?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2Nyb3NzY2x1c3Rlci9sb2dpY2FsOmxvZ2ljYWxfdGVzdA==) on master @ [533174dc4d87c792722bae844642f42dcf149aa7](https://github.com/cockroachdb/cockroach/commits/533174dc4d87c792722bae844642f42dcf149aa7):


```
        created by github.com/sasha-s/go-deadlock.lock in goroutine 16685428
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182 +0x14d
        
        goroutine 16685632 [runnable]:
        github.com/sasha-s/go-deadlock.lock.func1()
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182
        created by github.com/sasha-s/go-deadlock.lock in goroutine 16685428
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182 +0x14d
        
        goroutine 16686089 [runnable]:
        github.com/sasha-s/go-deadlock.lock.func1()
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182
        created by github.com/sasha-s/go-deadlock.lock in goroutine 12328989
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182 +0x14d
        
        goroutine 16685201 [runnable]:
        github.com/sasha-s/go-deadlock.lock.func1()
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182
        created by github.com/sasha-s/go-deadlock.lock in goroutine 16684915
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182 +0x14d
        
        goroutine 16685567 [runnable]:
        github.com/sasha-s/go-deadlock.lock.func1()
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182
        created by github.com/sasha-s/go-deadlock.lock in goroutine 16685428
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182 +0x14d
        
        goroutine 16686180 [runnable]:
        github.com/sasha-s/go-deadlock.lock.func1()
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182
        created by github.com/sasha-s/go-deadlock.lock in goroutine 22
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182 +0x14d
        
        goroutine 16686181 [runnable]:
        github.com/sasha-s/go-deadlock.lock.func1()
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182
        created by github.com/sasha-s/go-deadlock.lock in goroutine 22
        	github.com/sasha-s/go-deadlock/external/com_github_sasha_s_go_deadlock/deadlock.go:182 +0x14d
        
        goroutine 12318478 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/admission.(*tokenAllocationTicker).tick(...)
        	github.com/cockroachdb/cockroach/pkg/util/admission/io_load_listener.go:425
        github.com/cockroachdb/cockroach/pkg/util/admission.(*StoreGrantCoordinators).SetPebbleMetricsProvider.func1()
        	github.com/cockroachdb/cockroach/pkg/util/admission/grant_coordinator.go:113 +0xa5
        created by github.com/cockroachdb/cockroach/pkg/util/admission.(*StoreGrantCoordinators).SetPebbleMetricsProvider in goroutine 12307130
        	github.com/cockroachdb/cockroach/pkg/util/admission/grant_coordinator.go:107 +0x305
    logical_replication_job_test.go:622: timed out waiting for events
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestRandomStream3160439100
--- FAIL: TestRandomStream (52.98s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomStream.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/testutils/soon.go
+++ b/pkg/testutils/soon.go
@@ -17,6 +17,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/retry"
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/errors"
 )
@@ -82,7 +83,7 @@ func SucceedsWithinError(fn func() error, duration time.Duration) error {
 }
 
 func SucceedsSoonDuration() time.Duration {
-	if util.RaceEnabled {
+	if util.RaceEnabled || syncutil.DeadlockEnabled {
 		return RaceSucceedsSoonDuration
 	}
 	return DefaultSucceedsSoonDuration

